src/main/java/de/torui/coflsky/util/ServerSettingsLoader.java

```diff
@@ -0,0 +1,444 @@
+package de.torui.coflsky.util;
+
+import cc.polyfrost.oneconfig.libs.universal.UChat;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import de.torui.coflsky.config.CoflConfig;
+import de.torui.coflsky.config.SettingsCache;
+import net.minecraft.client.Minecraft;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.nio.ByteBuffer;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.*;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.core.Appender;
+import org.apache.logging.log4j.core.Filter;
+import org.apache.logging.log4j.core.Logger;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.appender.AbstractAppender;
+import org.apache.logging.log4j.core.layout.PatternLayout;
+
+/**
+ * Utility that, on request, executes the server command "/cofl get json", tails
+ * latest.log until the JSON blob is encountered, then applies those values to
+ * {@link CoflConfig} via reflection.
+ */
+public class ServerSettingsLoader {
+    private static final Path LOG_FILE = resolveLatestFile();
+    private static final Pattern JSON_PATTERN = Pattern.compile("\\[CoflSettings] (\\{.*})");
```

Comment:
don't parse console, hook into the eventbus with event [`OnSettingsReceive`](https://github.com/Coflnet/CoflSkyCore/blob/4a03709e6e320bb6d4a9df007ea9d04edfb40282/src/main/java/CoflCore/events/OnSettingsReceive.java#L5) instead, make sure to target the [branch its in](https://github.com/Coflnet/SkyblockMod/commits/CoreLib/)